addons/survey/models/survey_question.py

```diff
@@ -345,6 +346,33 @@ def _compute_is_scored_question(self):
             else:
                 question.is_scored_question = False
 
+    @api.depends('survey_id.session_speed_rating', 'is_time_limited', 'time_limit')
+    def _compute_is_time_customized(self):
+        for question in self:
+            if question.survey_id.session_speed_rating:
+                if question.time_limit != question.survey_id.session_speed_rating_time_limit:
+                    question.is_time_customized = True
+                else:
+                    question.is_time_customized = False
+            else:
+                question.is_time_customized = False
+
+    @api.depends('survey_id.session_speed_rating_time_limit', 'survey_id.session_speed_rating', 'is_time_limited')
+    def _compute_time_limit(self):
+        for question in self:
+            if not question.is_time_customized and question.survey_id.session_speed_rating:
+                question.time_limit = question.survey_id.session_speed_rating_time_limit
+            elif not question.is_time_limited or not question.survey_id.session_speed_rating:
+                question.time_limit = False
+
+    @api.depends('survey_id.session_speed_rating_time_limit', 'survey_id.session_speed_rating')
+    def _compute_is_time_limited(self):
+        for question in self:
+            if not question.is_time_customized and question.survey_id.session_speed_rating:
```

Comment:
here and above, it looks like  we're missing at least one default case. 
Then, you can also apply logic as in previous comment to simplify.

Reply:
since the values are stored and we need the compute function to pass if none of the above mentioned statements are true, we do not need to add a separate else case for this compute method.